Sources/PrimerSDK/Classes/Core/Payment Services/ClientTokenService.swift

```diff
@@ -83,35 +126,87 @@ internal class ClientTokenService: ClientTokenServiceProtocol {
             segments.append(tmpSecondSegment)
         }
         
-        let modifiedClientToken = segments.joined(separator: ".").base64RFC4648Format
+        return segments.joined(separator: ".").base64RFC4648Format
         
-        state.clientToken = modifiedClientToken
     }
+}
+
+extension ClientTokenService {
     
-    static func resetClientToken() {
+    // MARK: Store
+    
+    static func storeClientToken(_ clientToken: String) throws -> Promise<Void> {
+        return Promise { seal in
+            do {
+                try storeClientToken(clientToken) { error in
+                    if let error = error {
+                        seal.reject(error)
+                    } else {
+                        seal.fulfill()
+                    }
+                }
+            } catch {
+                seal.reject(error)
+            }
+        }
+    }
+    
+    static func storeClientToken(_ clientToken: String, completion: @escaping (Error?) -> Void) throws {
+        
         let state: AppStateProtocol = DependencyContainer.resolve()
-        state.clientToken = nil
+        
+        // 1. Validate the token manually or return the previous one from current App State
```

Comment:
Not entirely sure if this is ok. What's going to happen if the previous `clientToken` has been used (which means that it cannot be reused)?

Why not just reset it? The merchant provided us with an invalid token, why not just discard it?